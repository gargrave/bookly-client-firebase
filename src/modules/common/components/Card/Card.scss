@import '../../../../styles/_shadows';

$textColor: #6e6969;
$titleColor: #5d5656;

.card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
}

.hoverableCard:hover {
  background: #e4effc;
  cursor: pointer;
}

.header,
.title,
.text {
  margin: 0;
}

.header,
.title {
  color: $titleColor;
  font-weight: bold;
}

.header {
  font-size: 1.4rem;
}

.detailCard {
  padding: 15px;

  .title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .title + .cardText {
    font-size: 1.1rem;
  }

  .cardText + .cardText {
    margin-top: 8px;
  }
}

.listDetailCard {
  padding: 10px
}
