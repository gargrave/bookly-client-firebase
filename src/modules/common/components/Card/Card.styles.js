import styled from 'react-emotion'
import { colors, shadows } from '../../../../styles/'

export const StyledWrapper = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`
