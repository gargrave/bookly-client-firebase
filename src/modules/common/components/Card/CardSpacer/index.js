import CardSpacer from './CardSpacer'

export default CardSpacer
