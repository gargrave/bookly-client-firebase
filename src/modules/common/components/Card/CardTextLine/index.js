import CardTextLine from './CardTextLine'

export default CardTextLine
