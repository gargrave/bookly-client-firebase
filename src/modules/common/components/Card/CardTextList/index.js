import CardTextList from './CardTextList'

export default CardTextList
