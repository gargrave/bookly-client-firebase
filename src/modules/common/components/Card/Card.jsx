// @flow
import React from 'react'
import { any, bool, func } from 'prop-types'
import styled from 'react-emotion'

import { buildClass } from '../../../../utils/cssHelpers'

import CardHeader from './CardHeader'
import CardSpacer from './CardSpacer'
import CardTextLine from './CardTextLine'
import CardTextList from './CardTextList'

import { colors, shadows } from '../../../../styles/'

type Props = {
  children?: any,
  hoverable?: boolean,
  onClick?: Function,
}

const Styled = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
  text-align: center;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`

const Card = ({ children, hoverable = false, onClick }: Props) => {
  return (
    <Styled className={buildClass({ hoverable })} onClick={onClick}>
      {children}
    </Styled>
  )
}

Card.Header = CardHeader
Card.Spacer = CardSpacer
Card.TextLine = CardTextLine
Card.TextList = CardTextList

Card.propTypes = {
  children: any,
  hoverable: bool,
  onClick: func,
}

export default Card
