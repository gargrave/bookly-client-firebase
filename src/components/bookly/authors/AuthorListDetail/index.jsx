// @flow
import React from 'react';
import { func, shape, string } from 'prop-types';

import type { Author } from '../../../../constants/flowtypes';

import Card from '../../../common/Card';

type Props = {
  author: Author,
  onClick: Function,
};

function AuthorListDetail({
  author,
  onClick,
}: Props) {
  const authorName = `${author.firstName} ${author.lastName}`;

  return (
    <Card
      classes={['author-card']}
      onClick={onClick}
      title={authorName}
    />
  );
}

AuthorListDetail.propTypes = {
  author: shape({
    firstName: string.isRequired,
    lastName: string.isRequired,
  }),
  onClick: func.isRequired,
};

export default AuthorListDetail;
