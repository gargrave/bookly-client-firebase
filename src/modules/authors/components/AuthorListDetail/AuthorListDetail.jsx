// @flow
import React from 'react';
import { func, number, shape, string } from 'prop-types';

import type { Author } from '../../flowtypes';

import Card from '../../../common/components/Card/Card';

type Props = {
  author: Author,
  onClick: Function,
};

const bookCountString = (count?: number = 0): string => {
  const pluralizer = count === 1 ? '' : 's';
  return `${count} book${pluralizer}`;
};

const AuthorListDetail = ({
  author,
  onClick,
}: Props) => {
  const authorName = `${author.firstName} ${author.lastName}`;

  return (
    <Card
      classes={[
        'list-detail-card',
        'author-list-detail-card',
      ]}
      onClick={onClick}
    >
      <Card.TextLine bold text={authorName} />
      <Card.TextLine condensed text={bookCountString(author.bookCount)} />
    </Card>
  );
};

AuthorListDetail.propTypes = {
  author: shape({
    bookCount: number.isRequired,
    firstName: string.isRequired,
    lastName: string.isRequired,
  }),
  onClick: func.isRequired,
};

export default AuthorListDetail;
