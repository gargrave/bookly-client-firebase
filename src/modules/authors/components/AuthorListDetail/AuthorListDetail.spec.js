import { authorMocks } from '../../../../globals/mocks/'
import { ComponentBuilder } from '../../../../utils/testHelpers'

import Card from '../../../common/components/Card/Card'

import AuthorListDetail from './AuthorListDetail'

const defaultProps = {
  author: { ...authorMocks[0] },
  onClick: jest.fn(),
}

const builder = new ComponentBuilder(AuthorListDetail, defaultProps)

describe('AuthorListDetail', () => {
  let component

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent()
    expect(component).toMatchSnapshot()
  })

  it('renders correctly', () => {
    component = builder.shallowGetComponent()
    const element = component.find(Card)
    expect(element).toHaveLength(1)
  })
})
