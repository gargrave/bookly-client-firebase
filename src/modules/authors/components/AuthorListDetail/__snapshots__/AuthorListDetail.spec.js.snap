// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorListDetail matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorListDetail
    author={
      Object {
        "bookCount": 1,
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Aldous",
        "id": "oiw223h3qndlksadffl3ksdf",
        "lastName": "Huxley",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardTextLine
          bold={true}
          text="Aldous Huxley"
        />,
        <CardTextLine
          condensed={true}
          text="1 book"
        />,
      ],
      "classes": Array [
        "list-detail-card",
        "author-list-detail-card",
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bold": true,
          "text": "Aldous Huxley",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "condensed": true,
          "text": "1 book",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardTextLine
            bold={true}
            text="Aldous Huxley"
          />,
          <CardTextLine
            condensed={true}
            text="1 book"
          />,
        ],
        "classes": Array [
          "list-detail-card",
          "author-list-detail-card",
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bold": true,
            "text": "Aldous Huxley",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "condensed": true,
            "text": "1 book",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
